ui/src/actions/login.js

```diff
@@ -68,6 +69,12 @@ export function logIn(proposal, password) {
 
     dispatch(showErrorPanel(false));
     await dispatch(getLoginInfo());
+    const loginInfo = await fetchLoginInfo();
+    await dispatch(setLoginInfo(loginInfo));
+    if (loginInfo.limsName && loginInfo.limsName.length > 0) {
+      // synchronize the sample list with LIMS, Use First LIMS available as default
+      dispatch(syncSampleListWithLims(loginInfo.limsName[0].name));
+    }
```

Comment:
It would be good to not couple LIMS synchronisation with the login code.

When the user logs in, the `Main` component gets mounted and a `useEffect` invokes `dispatch(getInitialState());`. Perhaps this dispatch would be better placed inside this `getInitialState` action creator?

```js
export function getInitialState() {
  return async (dispatch, getState) => {
    const initialStateSlices = await Promise.all([...]);

    dispatch(setInitialState(Object.assign({}, ...initialStateSlices)));
    // HERE => call `getState` to get the LIMS name and dispatch `syncSampleListWithLims`
    dispatch(applicationFetched(true));
  };
}
```

Reply:
Hm, I'm wondering if its not worth investigating if the synchronization could not be done when the session is selected on the backend. There is something in place for handling the sessions selection/update, the synchronization could be done there and perhaps then simply fetched, right ?